pkg/operator/cluster.go

```diff
@@ -33,42 +33,58 @@ import (
 	kwatch "k8s.io/client-go/pkg/watch"
 )
 
-type clusterTPR struct {
+type clusterManager struct {
 	context      *context
+	name         string
 	watchVersion string
 	devicesInUse bool
 	clusters     map[string]*cluster.Cluster
 	tracker      *tprTracker
 	sync.RWMutex
+	// The initiators that create TPRs specific to specific Rook clusters.
+	// For example, pools, object services, and file services, only make sense in the context of a Rook cluster
+	inclusterInitiators []inclusterInitiator
+	inclusterMgrs       []tprManager
 }
 
-func newClusterTPR(context *context) *clusterTPR {
-	return &clusterTPR{
-		context:  context,
-		clusters: make(map[string]*cluster.Cluster),
-		tracker:  newTPRTracker(),
+func newClusterManager(context *context, inclusterInitiators []inclusterInitiator) *clusterManager {
+	return &clusterManager{
+		context:             context,
+		clusters:            make(map[string]*cluster.Cluster),
+		tracker:             newTPRTracker(),
+		inclusterInitiators: inclusterInitiators,
 	}
 }
 
-func (t *clusterTPR) Name() string {
+// Gets the name of the TPR
+func (t *clusterManager) Name() string {
 	return "rookcluster"
```

Comment:
is this the name of the object that manages all clusters?  where is this value used?  Is the hard coded value of "rookcluster" appropriate?